sdk/provisioning/Azure.Provisioning/tests/ProvisioningTests.cs

```diff
@@ -235,6 +235,34 @@ public async Task RedisCacheSecondaryConnectionString()
             await ValidateBicepAsync(interactiveMode: true);
         }
 
+        [RecordedTest]
+        public async Task RedisCacheWithExistingKeyVault()
+        {
+            TestInfrastructure infrastructure = new TestInfrastructure(configuration: new Configuration { UseInteractiveMode = true });
+            var cache = new RedisCache(infrastructure);
+            var kv = KeyVault.FromExisting(infrastructure, name: "existingVault");
```

Comment:
I left a comment above about refining this api.  It would be nice if it was `infra.AddExisting...`.  We can do that as a follow up to unblock the functionality.